src/model.cpp

```diff
@@ -120,6 +143,7 @@ Status Model::addVersions(std::shared_ptr<model_versions_t> versionsToStart, ovm
                 version,
                 status.string());
             result = status;
+            cleanupModelTmpFiles(config);
```

Comment:
If we add two versions during one reloadModelWithVersions call we will
download model files for both versions
then when 1 version will succeed to load & 2nd will fail (eg. missing .xml file), you remove local copy for version 1 as well. This cleanupModelTmpFiles should happen only for that specific version, or we should call downloadModels again for that specific version without changing localPath for that instance.

Reply:
Oh I see now you changed the implementation in cleanupTmpModelFiles from:
lfstatus = lfs.deleteFileFolder(config.getLocalPath());
To
lfstatus = lfs.deleteFileFolder(config.getPath());
Then it works.